pyflow/host.py

```diff
@@ -176,25 +176,23 @@ def check_cmd(self):
     def run_simple_command(self, cmd):
         raise NotImplementedError
 
-    @property
-    def preamble(self):
+    def preamble(self, exit_hook=[]):
         """*list*: The host-specific preamble script for jobs."""
         preamble = SET_ECF_VARIABLES.split("\n")
         if self.extra_paths:
             preamble.append("export PATH=%s:${PATH}" % (":".join(self.extra_paths),))
         for var, val in self.environment_variables.items():
             preamble.append('export {}="{}"'.format(var, val))
 
-        specific_preamble = self.host_preamble
+        specific_preamble = self.host_preamble(exit_hook)
         if specific_preamble:
             preamble += specific_preamble
         if self.extra_preamble:
             preamble.append("")
             preamble += self.extra_preamble
         return preamble
 
-    @property
-    def host_preamble(self):
+    def host_preamble(self, exit_hook=[]):
```

Comment:
If this method is meant to be overridden, we should definitely not encourage using lists as default arguments